src/engine/physics/physics_system.cppm

```diff
@@ -0,0 +1,450 @@
+                backend_->SetRigidBodyTransform(entity, transform);
+            }
+        }
+
+        // Get rigid body position
+        [[nodiscard]] glm::vec3 GetPosition(EntityId entity) const {
+            if (backend_) {
+                return backend_->GetRigidBodyTransform(entity).position;
+            }
+            return glm::vec3{0.0F};
+        }
+
+        // Get rigid body rotation
+        [[nodiscard]] glm::quat GetRotation(EntityId entity) const {
+            if (backend_) {
+                return backend_->GetRigidBodyTransform(entity).rotation;
+            }
+            return glm::quat{1.0F, 0.0F, 0.0F, 0.0F};
+        }
+
+        // Set linear velocity
+        void SetLinearVelocity(EntityId entity, const glm::vec3& velocity) {
+            if (backend_) {
+                backend_->SetLinearVelocity(entity, velocity);
+            }
+        }
+
+        // Get linear velocity
+        [[nodiscard]] glm::vec3 GetLinearVelocity(EntityId entity) const {
+            return backend_ ? backend_->GetLinearVelocity(entity) : glm::vec3{0.0F};
+        }
+
+        // Set angular velocity
+        void SetAngularVelocity(EntityId entity, const glm::vec3& velocity) {
+            if (backend_) {
+                backend_->SetAngularVelocity(entity, velocity);
+            }
+        }
+
+        // Get angular velocity
+        [[nodiscard]] glm::vec3 GetAngularVelocity(EntityId entity) const {
+            return backend_ ? backend_->GetAngularVelocity(entity) : glm::vec3{0.0F};
+        }
+
+        // Apply force at center of mass
+        void ApplyForce(EntityId entity, const glm::vec3& force) {
+            if (backend_) {
+                backend_->ApplyForce(entity, force);
+            }
+        }
+
+        // Apply force at position
+        void ApplyForceAtPosition(EntityId entity, const glm::vec3& force, const glm::vec3& position) {
+            if (backend_) {
+                backend_->ApplyForceAtPosition(entity, force, position);
+            }
+        }
+
+        // Apply impulse at center of mass
+        void ApplyImpulse(EntityId entity, const glm::vec3& impulse) {
+            if (backend_) {
+                backend_->ApplyImpulse(entity, impulse);
+            }
+        }
+
+        // Apply impulse at position
+        void ApplyImpulseAtPosition(EntityId entity, const glm::vec3& impulse, const glm::vec3& position) {
+            if (backend_) {
+                backend_->ApplyImpulseAtPosition(entity, impulse, position);
+            }
+        }
+
+        // Apply torque
+        void ApplyTorque(EntityId entity, const glm::vec3& torque) {
+            if (backend_) {
+                backend_->ApplyTorque(entity, torque);
+            }
+        }
+
+        // === Collider Management ===
+
+        // Add a collider
+        bool AddCollider(EntityId entity, const ColliderConfig& config = ColliderConfig{}) {
+            return backend_ && backend_->AddCollider(entity, config);
```

Comment:
A few of these methods using an if (backend_) while other use a short-circuit backend_ &&, and even others use a ternary. Make it consistent